### PR TITLE
Implement hypershard-android

### LINKED ISSUE #606: Android Hypershard example
Provide an example of using Hypershard for companies that are unable to use dex parsing on Android.

- https://github.com/dropbox/hypershard-android

The example should be described step by step in the markdown document.

### PR DESCRIPTION
**Pull request by pawelpasterz**
Fixes #606

## Checklist

- [ ] Documented
- [ ] Unit tested
- [ ] release_notes.md updated

### CODE REVIEW

**Comment by bootstraponline** [comment]
I updated the tickets.

- #606 hypershard for Android
- #766 hypershard for iOS

File: test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    parameterListHeading = "%n@|bold,underline Parameters:|@%n",
+    optionListHeading = "%n@|bold,underline Options:|@%n",
+    header = ["Run hypershard-android against test classes"],
+    description = ["Hypershard is a fast and simple test collector that uses the Kotlin and Java ASTs. " +
+            "Flank will save to file full qualified test names found in dir(s)" +
+            " that can be used later as input for --test-targets option."
+    ],
+    usageHelpAutoWidth = true
+)
+class HyperAndroidCommand : Runnable {
```
Comment (bootstraponline) [inline test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt:34]: Let's include an example in the readme.

**Comment by bootstraponline** [comment]
What's the proper syntax for running the command?

```
$ flank hypershard android ./flaky-androidTest.apk

Unmatched argument at index 2: './flaky-androidTest.apk'
```

I expected this to create a text file with a list of tests.

**Comment by bootstraponline** [comment]
The code looks great. I pointed it at the src folder. The only weird thing is `Done!ng` seems like a typo?

```
$ flank hypershard android --dirs ./app/src/androidTest
version: local_snapshot
revision: 5b3529759441998591d0b8cda02237673e14001d

Done!ng: ⡿
Results file writen to ~/flank/test_app/hypershard-android-results.txt
```

File: test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            val tests = hyperShard.gatherTests().map { it.replace(regex, "#") }
+
+            File(resultPath).bufferedWriter().use { out ->
+                tests.forEach {
+                    out.write(it)
+                    out.newLine()
+                }
+                spinner.cancel()
+                println("\rDone!")
+                println("Results file writen to $resultPath")
```
Comment (bootstraponline) [inline test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt:55]: Let's print how many tests are written to this file. Something like:

```kotlin
println("${tests.length} tests written to $resultPath")
```

File: test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt (hunk 1 above)
Comment (pawelpasterz) [reply test_runner/src/main/kotlin/ftl/cli/hypershard/android/HyperAndroidCommand.kt:34]: Definitely, this PR is still in progress.

**Comment by pawelpasterz** [comment]
Closing this PR since there are some obstacles that need to be solved first.

**Pull request closed without merging**